classes/validation/ValidatorUrl.inc.php

```diff
@@ -14,39 +14,20 @@
  * @brief Validation check for URLs.
  */
 
-import('lib.pkp.classes.validation.ValidatorUri');
+import ('lib.pkp.classes.validation.Validator');
+import('lib.pkp.classes.validation.ValidatorFactory');
 
-class ValidatorUrl extends ValidatorUri {
+class ValidatorUrl extends Validator {
 	/**
-	 * Constructor.
+	 * @copydoc Validator::isValid()
 	 */
-	function __construct() {
-		parent::__construct(ValidatorUrl::_getAllowedSchemes());
-	}
+	function isValid($value) {
+		$validator = \ValidatorFactory::make(
+			['value' => $value],
+			['value' => 'url']
+		);
 
-	//
-	// Public static methods
-	//
-	/**
-	 * @see ValidatorUri::getRegexp()
-	 * @param $allowedSchemes Array of strings to restrict accepted schemes to defined set, or null for any allowed
-	 * @return string
-	 */
-	static function getRegexp($allowedSchemes = null) {
-		if ($allowedSchemes === null) $allowedSchemes = self::_getAllowedSchemes();
-		else $allowedSchemes = array_intersect(self::_getAllowedSchemes(), $allowedSchemes);
-		return parent::getRegexp($allowedSchemes);
-	}
-
-	//
-	// Private static methods
-	//
-	/**
-	 * Return allowed schemes
-	 * @return array
-	 */
-	static function _getAllowedSchemes() {
-		return array('http', 'https', 'ftp');
+		return $validator->passes();
```

Comment:
Incidentally, does this resolve https://github.com/pkp/pkp-lib/issues/3855?

Reply:
No, but [this does](https://github.com/NateWr/pkp-lib/blob/i3594_forms/classes/validation/ValidatorFactory.inc.php#L44-L64).

Reply:
Bonus! Feel free to close that one when this gets merged.